mmcls/models/necks/gem.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch
+from torch import Tensor, nn
+from torch.nn import functional as F
+from torch.nn.parameter import Parameter
+
+from ..builder import NECKS
+
+
+def gem(x: Tensor, p: int = 3, eps: float = 1e-6) -> Tensor:
+    return F.avg_pool2d(x.clamp(min=eps).pow(p),
+                        (x.size(-2), x.size(-1))).pow(1. / p)
+
+
+@NECKS.register_module()
+class GeneralizedMeanPooling(nn.Module):
+    """Generalized Mean Pooling neck.
+
+    Note that we use `view` to remove extra channel after pooling. We do not
+    use `squeeze` as it will also remove the batch dimension when the tensor
+    has a batch dimension of size 1, which can lead to unexpected errors.
+
+    Args:
+        p:
+            Parameter value
+        eps:
+            epsilon
+    """
+
+    def __init__(self, p: int = 3, eps: float = 1e-6):
```

Comment:
`p` can be a `float`, in [the official repo. of multigrain](https://github.com/facebookresearch/multigrain/blob/172bb2a2ef8a7f69f02fd6a4eb20c9c3d5d7a909/multigrain/modules/functional.py#L30), `p` can be `math.inf`, which is a float.

It will be better to add an `argument` of `clamp` like https://github.com/facebookresearch/multigrain/blob/172bb2a2ef8a7f69f02fd6a4eb20c9c3d5d7a909/multigrain/modules/functional.py#L30